solr/core/src/java/org/apache/solr/handler/RequestHandlerBase.java

```diff
@@ -250,17 +253,17 @@ public void handleRequest(SolrQueryRequest req, SolrQueryResponse rsp) {
       long elapsed = timer.stop();
       metrics.totalTime.inc(elapsed);
 
-      if (cpuStats != null) {
-        Optional<Long> cpuTime = cpuStats.getCpuTimeMs();
+      if (publishCpuTime) {
```

Comment:
It looks like the present implementation cannot avoid logging cpu time if cpuAllowed is used?

Reply:
I don't think that's the case - see above comment.